saleor/graphql/product/tests/test_product_channel_listing_update.py

```diff
@@ -988,3 +1002,227 @@ def test_product_channel_listing_update_publish_product_without_category(
     assert errors[0]["code"] == ProductErrorCode.PRODUCT_WITHOUT_CATEGORY.name
     assert errors[0]["channels"] == [channel_usd_id]
     assert len(errors) == 1
+
+
+def test_product_channel_listing_add_variant_as_staff_user(
+    staff_api_client, product, permission_manage_products, channel_USD
+):
+    # given
+    product.variants.all().delete()
+    variant_1 = ProductVariant.objects.create(product=product, sku="321")
+    variant_2 = ProductVariant.objects.create(product=product, sku="333")
+    variant_1_id = graphene.Node.to_global_id("ProductVariant", variant_1.pk)
+    variant_2_id = graphene.Node.to_global_id("ProductVariant", variant_2.pk)
+    variants = [variant_1_id, variant_2_id]
+
+    product_id = graphene.Node.to_global_id("Product", product.pk)
+    channel_id = graphene.Node.to_global_id("Channel", channel_USD.id)
+    variables = {
+        "id": product_id,
+        "input": {
+            "updateChannels": [{"channelId": channel_id, "addVariants": variants}]
+        },
+    }
+
+    # when
+    response = staff_api_client.post_graphql(
+        PRODUCT_CHANNEL_LISTING_UPDATE_MUTATION,
+        variables=variables,
+        permissions=(permission_manage_products,),
+    )
+    content = get_graphql_content(response)
+
+    # then
+    data = content["data"]["productChannelListingUpdate"]
+    variant_data = data["product"]["variants"]
+
+    assert not data["productChannelListingErrors"]
+    assert variant_data[0]["channelListings"][0]["channel"]["slug"] == channel_USD.slug
+    assert variant_data[1]["channelListings"][0]["channel"]["slug"] == channel_USD.slug
```

Comment:
Could you also check if `variants` have been correctly added? Please, check the same in the other tests.